tao_compiler/mlir/disc/transforms/dot_rewriter.cc

```diff
@@ -395,15 +395,6 @@ struct TransposeFoldingConvert : public OpRewritePattern<mhlo::DotGeneralOp> {
     Value dot = rewriter.create<mhlo::DotGeneralOp>(
         loc, op.getType(), lhs, rhs, dot_dimension_attr, nullptr);
     rewriter.replaceOp(op, dot);
-
-    // Remove transpose op which outputs into dot.
-    if (tp_lhs) {
-      rewriter.eraseOp(old_lhs.getDefiningOp());
```

Comment:
Maybe we can add a filecheck based test? For example:

```
%0 = transpose // unsafe to remove transpose directly
%1 = matmul(%0, ...)
return %0, %1
```

Reply:
done